src/actions/sources/newSources.js

```diff
@@ -44,15 +44,20 @@ function createOriginalSource(
  * @memberof actions/sources
  * @static
  */
-export function loadSourceMap(generatedSource: Source) {
+export function loadSourceMap(generatedSourceId: SourceId) {
   return async function({ dispatch, getState, sourceMaps }: ThunkArgs) {
-    let urls;
+    const generatedSource = getSource(getState(), generatedSourceId);
```

Comment:
the reason why the source state reverted to the loading state (i believe) is that we were using stale source data.

Re-querying the store helps with that